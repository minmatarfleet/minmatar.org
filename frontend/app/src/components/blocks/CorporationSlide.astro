---
import { i18n } from '@helpers/i18n'
const { t, translatePath } = i18n(Astro.url)

import type { CorporationObject } from '@dtypes/layout_components'

interface Props {
    id:                     string;
    ceo_image:              string;
    ceo_character_id:       number;
    corporation:            CorporationObject;
    is_user_corporation?:   boolean;
    picture_placement:      'left' | 'right';
    mirror_slide?:          boolean;
}

const {
    id,
    ceo_image,
    ceo_character_id,
    corporation,
    is_user_corporation = false,
    picture_placement = 'left',
    mirror_slide = false,
} = Astro.props

import FixedFluid from "@components/compositions/FixedFluid.astro";
import Flexblock from "@components/compositions/Flexblock.astro";

import TextBox from '@components/layout/TextBox.astro';

import VerticalCenter from "@components/blocks/VerticalCenter.astro";
import CorporationSlideStatus from "@components/blocks/CorporationSlideStatus.astro";
import StylessButton from '@components/blocks/StylessButton.astro';

import { get_corporation_logo } from '@helpers/eve_image_server';
import { query_string } from '@helpers/string';
import Wrapper from '@components/compositions/Wrapper.astro';
---

<FixedFluid
    id={id}
    class:list={[ 'corporation-intro',  'w-full items-center', { mirrorwed: mirror_slide } ]}
    width='650px'
    breakpoint='30%'
    fluid_first={picture_placement === 'left'}
>
    <Wrapper
        padding_block='var(--space-s)'
        padding_inline='var(--space-s)'
    >
        <Flexblock gap='var(--space-xl)' class="[ content ][ pb-32 lg:pb-0 ]">
            <VerticalCenter>
                <FixedFluid width='64px' class="[ items-center w-full ]">
                    <picture class="[ border ]">
                        <img loading="lazy" src={get_corporation_logo(corporation.corporation_id, 64)} width="64" height="64" alt={`${corporation.corporation_name} ${t('corporation_logo')}`} />
                    </picture>
                    <h2>{corporation.corporation_name}</h2>
                </FixedFluid>
            </VerticalCenter>
            <TextBox>
                <slot />
            </TextBox>
            <CorporationSlideStatus
                corporation={corporation}
                is_user_corporation={is_user_corporation}
            />
        </Flexblock>
    </Wrapper>
    
    <div class="[ corporation-identity ]">
        <picture
            class:list={{ 'ceo-right': picture_placement === 'right' }}
        >
            <StylessButton
                class="[ ceo ]"
                x-data={`{
                    show_character_modal() {
                        show_modal({
                            partial: '${translatePath('/partials/character_modal_component/')}?${query_string({
                                character_id: ceo_character_id.toString(),
                            })}',
                        })
                    }
                }`}
                x-on:click="show_character_modal()"
            >
                <img src={ceo_image} height="496" width="940" />
            </StylessButton>
            <slot name="coceo" />
            <slot name="ships" />
        </picture>
    </div>
</FixedFluid>

<style lang="scss">
    .corporation-intro {
        position: relative;
    }

    h2 {
        font-size: var(--step-2);
    }

    img {
        image-rendering: pixelated;
    }

    .corporation-identity {
        width: 100%;
        justify-content: center;
        display: flex;

        @media screen and (max-width: 1279px) {
            display: none;
        }
    }

    picture {
        aspect-ratio: 1 / 1;
        max-width: 900px;
        position: relative;
        width: 100%;
        display: flex;

        &.ceo-right {
            justify-content: flex-end;
        }
    }

    .ceo {
        z-index: 10;
        position: relative;
        cursor: pointer;
        transform: scale(0.5);
        opacity: 0;
        height: 99%;

        img {
            mask-image: linear-gradient(to bottom, black 70%, transparent 95%);
            width: auto;
            height: 100%;
        }

        &:focus-visible {
            outline: none;
        }

        &:focus-visible,
        &:hover {
            img {
                filter: drop-shadow(0px 0px 3px rgba(255, 255, 255, 0.75));
            }
        }
    }

    @keyframes ceo {
        from {
            transform: scale(0.5);
            opacity: 0;
        }

        to {
            transform: scale(1);
            opacity: 1;
        }
    }
</style>

<style lang="scss" is:global>
    .mirrorwed {
        transform: scaleX(-1);

        .content > *:not(.co-ceo) {
            transform: scaleX(-1);
        }
    }

    .swiper-slide-active {
        .ceo {
            animation: ceo ease-out 1.5s;
            animation-delay: 0.5s;
            animation-iteration-count: 1;
            transform-origin: bottom;
            animation-fill-mode: forwards;
        }
    }
</style>