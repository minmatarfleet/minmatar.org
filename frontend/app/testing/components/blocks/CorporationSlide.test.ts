import { experimental_AstroContainer as AstroContainer } from "astro/container";
import { expect, test, vi } from "vitest";
import CorporationSlide from "@components/blocks/CorporationSlide.astro";

test("CorporationSlide defaults", async () => {
  const container = await AstroContainer.create();
  const result = await container.renderToString(CorporationSlide, {
    props: {
      id: 65,
      ceo_image: "chicken.gif",
      ceo_character_id: 89,
      corporation: {
        corporation_id: 65,
        corporation_name: "chicken attack",
      },
    },
    slots: {
      default: "slotted chick",
    },
  });

  // expect(result).toMatchSnapshot();
});
