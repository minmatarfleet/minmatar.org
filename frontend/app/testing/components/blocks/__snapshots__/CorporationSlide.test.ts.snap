// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CorporationSlide defaults 1`] = `
"<div class="fixed-fluid fluid_first corporation-intro w-full items-center" id="65" data-astro-cid-c2ktugqm="true" data-mod data-astro-cid-34go4nvk style="--fixed_fluid_gap: var(--space-s);--fixed_fluid_width: 650px;--fixed_fluid_breakpoint: 30%;">  <div class="wrapper" data-astro-cid-c2ktugqm="true" data-mod data-astro-cid-yv35cmqx style="--wrapper_padding_inline: var(--space-s);--wrapper_padding_inline_end: var(--space-s);--wrapper_padding_block: var(--space-s);--wrapper_max_width: 80rem;">  <div class="flex-block [ content ][ pb-32 lg:pb-0 ] gap---space-xl" data-astro-cid-c2ktugqm="true" data-mod style="--flex_block_gap: var(--space-xl);">  <div class="!flex !items-center" data-astro-cid-c2ktugqm="true">  <div class="fixed-fluid [ items-center w-full ]" data-astro-cid-c2ktugqm="true" data-mod data-astro-cid-34go4nvk style="--fixed_fluid_gap: var(--space-s);--fixed_fluid_width: 64px;--fixed_fluid_breakpoint: 50%;">  <picture class="[ border ]" data-astro-cid-c2ktugqm> <img loading="lazy" src="https://images.evetech.net/corporations/65/logo?size=64" width="64" height="64" alt="chicken attack corporation logo" data-astro-cid-c2ktugqm> </picture> <h2 data-astro-cid-c2ktugqm>chicken attack</h2>  </div>    </div> <div class="wrapper" data-mod data-astro-cid-yv35cmqx style="--wrapper_padding_inline: 0;--wrapper_padding_inline_end: 0;--wrapper_padding_block: 0;--wrapper_max_width: var(--max-text-width);">   slotted chick   </div>   <div class="flex-block gap---space-xl" id="corporation-status-65" x-data="{
        init() {
            
        }
    }" data-astro-cid-ggpwwpcg="true" data-mod style="--flex_block_gap: var(--space-xl);">  <div class="flex-inline" data-astro-cid-ggpwwpcg="true" data-astro-cid-ud5fqf5k style="--flex_inline_gap: var(--space-s);--flex_inline_justification: flex-start;"> <button x-data="{
            show_corporation_modal() {
                    show_modal({
                        partial: '/partials/corporation_modal_component/?corporation_id=65',
                    })
                }
            }" x-on:click.stop.prevent="show_corporation_modal()" data-astro-cid-ggpwwpcg="true" data-size="sm" style="--button_background_color: var(--fleet-red); --button_color: var(--fleet-yellow)" disabled x-init="$el.removeAttribute(&#34;disabled&#34;)" data-astro-cid-r5e2wo56="true" class="[ button ]"> <div class="flex-inline" data-astro-cid-r5e2wo56="true" data-astro-cid-ud5fqf5k style="--flex_inline_gap: var(--space-xs);--flex_inline_justification: center;">   Learn more  </div>   </button>   </div>    </div>    </div>   </div>   <div class="[ corporation-identity ]" data-astro-cid-c2ktugqm> <picture data-astro-cid-c2ktugqm> <button class="[ ceo ]" x-data="{
                    show_character_modal() {
                        show_modal({
                            partial: '/partials/character_modal_component/?character_id=89',
                        })
                    }
                }" x-on:click="show_character_modal()" data-astro-cid-c2ktugqm="true" data-astro-cid-syldxee3="true">  <img src="chicken.gif" height="496" width="940" data-astro-cid-c2ktugqm>  </button>    </picture> </div>  </div>    "
`;
